qtfred/src/mission/Editor.cpp

```diff
@@ -3118,5 +3110,71 @@ int Editor::global_error_check_mixed_player_wing(int w) {
 
 	return 0;
 }
+
+bool Editor::compareShieldSysData(const std::vector<int>& teams, const std::vector<int>& types) const {
+	Assert(Shield_sys_teams.size() == teams.size());
+	Assert(Shield_sys_types.size() == types.size());
+	if (Shield_sys_teams.size() != teams.size() || Shield_sys_types.size() != types.size()) {
```

Comment:
This check is unnecessary. This only need this `if` statement or the `Assert` not both.

Reply:
I'm not so sure it's unreasonable to have both checks. The asserts will fail noisily in debug mode and the if-check will fail silently in release mode. That said, if you want only one of the two, I'll take the asserts since at least those will fail noisily.

Reply:
Well, the idea behind an assert is that it checks something that should always be true. If you then repeat the check outside of an `Assert` statement it means that the condition may be false in some circumstances which means that there may be a coding error somewhere.